pkg/recycle/drain.go

```diff
@@ -50,7 +51,35 @@ func (r *Recycler) drainNode(helper *drain.Helper) error {
 		return err
 	}
 
-	return drain.RunNodeDrain(helper, r.nodeToRecycle.Name)
+	err = drain.RunNodeDrain(helper, r.nodeToRecycle.Name)
+	if err != nil {
+		pendingList, newErrs := helper.GetPodsForDeletion(r.nodeToRecycle.Name)
+		if pendingList != nil {
+			pods := pendingList.Pods()
+			if len(pods) != 0 {
+				log.Warn().Msgf("There are pending pods to drain on node %q - Retrying with disable-eviction enabled.",
+					r.nodeToRecycle.Name)
+				for _, pendingPod := range pods {
+					log.Warn().Msgf("%s/%s\n", "pod", pendingPod.Name)
+				}
+				// retry deleting the pending pods with disable-eviction enabled
+				// This wll delete the pods instead of evicting
+				helper.DisableEviction = true
+				err = drain.RunNodeDrain(helper, r.nodeToRecycle.Name)
```

Comment:
Would this look nicer if we require the drain.RunNodeDrain method argument `disableEviction` or something of that nature. Just trying to make the code flow a bit.

Reply:
`RunNodeDrain` is the external function from kubectl drain package. We cant add argument unless we want to copy the function and write our own. is that what you meant?

Reply:
Ofc it is. Ignore me.